tests/FixerNameValidatorTest.php

```diff
@@ -50,15 +50,25 @@ public static function provideIsValidCases(): array
             ['_foo', false, false],
             ['4_foo', false, false],
             ['vendor/foo', false, false],
-            ['bendor/foo', true, false],
+            ['vendor/foo', true, true],
             ['Vendor/foo', true, true],
             ['Vendor4/foo', true, true],
             ['4vendor/foo', true, false],
             ['Vendor/foo', true, true],
             ['FooBar/foo', true, true],
+            ['foo\\bar', true, true],
```

Comment:
original idea was to support format of `vendor/rule`. now you also add `vendor\rule`.
not sure if I would like to keep 2, `/` and `\`, alternative syntaxes
I would like to keep explicit `vendor/` part - and then whatever after

Reply:
@keradus did you read #7062 as a background for this change? This is whole point of this PR - to allow FQCNs as rule name. Right now it's impossible to use rules from `symplify/coding-standard` and there are several other drawbacks of using "random" strings.

Reply:
if we expose our integration point with interface and we expect specific format, ie `vendor/rule`, and some external library decides to not use that format, I would rather expect that format that is expected is provided, rather than we change expectations.
^^ that's about the principle approach in general.

### point 1
for this case specifically, i would love to see have single way how we format of rules. and if it's sometimes `foo/bar`, and sometimes `foo\bar`, then it's off and I would love to avoid it.

### point 2
if we consider to touch possible format of name for custom rules, i would suggest to combine the alignment with possible format of name for custom rulesets

### point 3
if using random string is a drawback, let's solve that drawback properly - change of string format is not solving them

Reply:
> if we expose our integration point with interface and we expect specific format, ie vendor/rule, and some external library decides to not use that format, I would rather expect that format that is expected is provided, rather than we change expectations.

This PR is not created in order to align with userland convention, but to address several concerns raised in #7062. So this is not about changing expectation, rather about changing general approach. Using FQCNs enables several advantages - code navigation, usage detection, visual deprecation indicator, easy refactoring etc.

In terms of other points:

1. I think this can be only for transitional period, because I think FQCN would be better as a contract in the future. At this point we can allow both and systematically, step by step, shift in FQCN direction. `vendor/name` is kept for backward compatibility (allows only 1 `/something` segment), while newly introduced `vendor\name` allow 1+ `\something` segment(s) and is forward-compatible. 
2. Yes, we could allow FQCNs as a name of custom rule sets too.
3. Not sure what you mean. We can't solve it just like that, as I wrote in the point 1 - it requires transition period. Allowing FQCNs is first step of the process which may lead to full switch to FQCNs in v4.

PS. Support for FQCNs for built-in fixers can be done with 2 lines, by adding `$this->fixersByName[get_class($fixer)] = $fixer;` to `FixerFactory::registerFixer()` and aligning `$availableFixers` in `ConfigurationResolver::validateRules()`. I could even add it in this PR so people can instantly start configuring Fixer using FQCNs, at least when it comes to fixers, not rulesets.